spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -306,7 +306,7 @@ static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
+				return (originalContentType.startsWith(MimeTypeUtils.APPLICATION_JSON_VALUE)) ? MimeTypeUtils.APPLICATION_JSON
```

Comment:
Actually I take that back.

It would be better overall for this to be explained with a test case.

Even if `contentType` is set to `text/plain`, `originalContentType` remains `application/json;charset=UTF-8`. Since on receiving the message, the binder uses `originalContentType` to replace `contentType`, it will handle a message that effectively has the charset-bearing version as `contentType`. The `contentType` header just instructs a binder that understands headers (e.g. Rabbit) that the content of the message is textual. The special treatment of JSON allows non-Spring Cloud Stream applications to handle the content as JSON. Overall, it is better to preserve the content type, but it only affects non-SCSt consumers.